tests/test_transform.py

```diff
@@ -170,19 +171,26 @@ def test_apply_wind_vane_dead_band_offset():
             data['Dir78mS'].fillna(0).round(10)).all()
 
 
-def test_freq_str_to_timedelta():
-    periods = ['1min', '5min', '10min', '15min',
-               '1H', '3H', '6H',
-               '1D', '7D', '1W', '2W',
-               '1MS', '1M', '3M', '6MS',
+def test_freq_str_to_dateoffset():
+    # Excluding monthly periods and above as it will depend on which month or year
+    periods = ['1S', '1min', '5min', '10min', '15min',
+               '1H', '3H', '6H', '1D', '7D',
+               '1W', '2W', '1MS', '1M', '3M', '6MS',
                '1AS', '1A', '3A']
-    results = [60.0, 300.0, 600.0, 900.0,
-               3600.0, 10800.0, 21600.0,
-               86400.0, 604800.0, 604800.0, 1209600.0,
-               2629746.0, 2629746.0, 7889238.0, 15778476.0,
-               31536000.0, 31536000.0, 94608000.0]
+    results = [1.0, 60.0, 300.0, 600.0, 900.0,
+               3600.0, 10800.0, 21600.0, 86400.0, 604800.0,
+               604800.0, 1209600.0, 2678400.0, 2678400.0, 7862400.0, 15724800.0,
+               31622400.0, 31622400.0, 94694400.0]
+
     for idx, period in enumerate(periods):
-        assert bw.transform.transform._freq_str_to_timedelta(period).total_seconds() == results[idx]
+        if type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset:
+            # The data frequency is returned as a DateOffset. The time delta can be in seconds
+            # can be derived adding the date offset to an actual date.
+            assert (ref_date + bw.transform.transform._freq_str_to_dateoffset(period) - ref_date
+                    ).total_seconds() == results[idx]
+
+    # Check that data frequency is returned as a DateOffset.
+    assert type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset
 
```

Comment:
```
# Check that data frequency is returned as a DateOffset.
assert type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset
```
@BiancaMorandi These 2 lines should be tabbed in one and be within the for loop.